### PR TITLE
Test coverage for Gateways Controllers

### PR DESCRIPTION
**Pull request by gterzian**
Work in progress

I'll also try to write a failing test for https://github.com/stellar/stellar-client/issues/1166 (and perhaps point to a fix)

### CODE REVIEW

File: app/scripts/services/gateways-service.js [hunk 1]
```diff
@@ -47,6 +47,7 @@ sc.service('Gateways', function($q, $analytics, session, StellarNetwork, rpStell
 
   Gateways.remove = function(gateway) {
     gateway.status = "removing";
+    walletGateways()[gateway.domain].status = "removing";
```
Comment (gterzian) [inline app/scripts/services/gateways-service.js]: just made this one change to the service, actually this just matches what goes on in the 'add' method. Without this the tests would not pass (unless I would already set the status of the gateway in the session to 'removing')

**Comment by gterzian** [comment]
@nullstyle Ok this one is good to go on my side. Just made on change to a controller (see line comment at https://github.com/stellar/stellar-client/pull/1167#discussion-diff-25132219).

**Comment by nullstyle** [comment]
:+1:

**Comment by bekkibolthouse** [comment]
@gterzian thanks!

**Comment by nullstyle** [comment]
Whoops, yeah I forgot to send thanks.  Thanks @gterzian!

**Pull request merged**